wicker/core/datasets.py

```diff
@@ -303,3 +452,85 @@ def dataset_size(self) -> int:
             int: total dataset size in bytes
         """
         return self._get_dataset_partition_size()
+
+
+class DatasetFactory:
+    """Provides functions to make it easier for users to create instances of Wicker datasets by
+    creating them with standardized parameters.
+
+    To customize the dataset creation process, we recommended deriving a new subclass of this
+    factory and overriding its methods (or adding new ones) in order to support the specific kind of
+    Wicker dataset that you want.
+    """
+
+    def create_filesystem_dataset(
```

Comment:
Can we just expose a create dataset method and return the abstract dataset interface?